### DISCUSSION TITLE
Day time dissolved data

### DISCUSSION THREAD

**Discussion by Luk5665** [thomluther/anker-solix-api#140] [Q&A] [answered]
Hi, I have tried out this python package today's evening. First of all, thanks for editing this package.
During my experimental session I have used the script for exporting the key Solarbank parameters during a time period of days to CSV file. However, I am interested in the daytime dissolved data of multiple days. Have someone creating a script for doing this? And at which time-scale you are possible to extract data, a couple of seconds or minutes?

I would be glad if someone has already edited such a script. :)

**Comment by thomluther** [comment, marked as answer]
You have to test that on your own, the tool was only made to extract data on a daily base and that is already lots of queries since some data is only provided for the total queried interval, but not for each slot.
You can use the energy_analysis helper method of the AnkerSolixApi class
```
async def energy_analysis(
    self,
    siteId: str,
    deviceSn: str,
    rangeType: str | None = None,
    startDay: datetime | None = None,
    endDay: datetime | None = None,
    devType: str | None = None,
) -> dict:
    """Fetch Energy data for given device and optional time frame.

    siteId: site ID of device
    deviceSn: Device to fetch data # This does not really matter since system level data provided, but field is mandatory
    rangeType: "day" | "week" | "year"
    startTime: optional start Date and time
    endTime: optional end Date and time
    devType: "solar_production" | "solar_production_pv[1-4]" | "solarbank" | "home_usage" | "grid" 
```
daytime data can only be extracted for full days, I believe they are in 10 min intervals or even longer...
